app/helpers/application_helper.rb

```diff
@@ -135,4 +135,8 @@ def storage_location_for_source(source_object)
     end
     current_organization.default_storage_location
   end
+
+  def filter_inactive_locations(storage_locations)
+    storage_locations.select { |storage_location| !storage_location.discarded_at }
```

Comment:
This seems like it should be a scope on `storage_locations` (and part of the query) rather than a method that filters in memory.